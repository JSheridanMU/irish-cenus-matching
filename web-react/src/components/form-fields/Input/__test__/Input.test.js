import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Input from '.././Input'

afterEach(cleanup)

test('renders without crashing', () => {
  render(<Input label="Test Label" />)
})

test('renders button with text', () => {
  const { getAllByText } = render(<Input label="Test Label" />)

  getAllByText('Test Label')
})

test('matches snapshot', () => {
  const tree = renderer.create(<Input label="Test Label" />).toJSON()
  expect(tree).toMatchSnapshot()
})
