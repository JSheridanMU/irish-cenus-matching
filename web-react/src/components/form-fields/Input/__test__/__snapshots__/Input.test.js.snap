// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
  >
    Test Label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-11"
      >
        <span>
          Test Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
